src/Platform/Microsoft.Testing.Platform/Messages/ListTestsMessageBus.cs

```diff
@@ -70,6 +75,11 @@ public override async Task PublishAsync(IDataProducer dataProducer, IData data)
             return;
         }
 
+        if (_dotnetTestConnection?.IsConnected == true && _dotnetTestDataConsumer is not null)
```

Comment:
do we need both here? Aren't "aligned", I mean if _dotnetTestConnection?.IsConnected == true the consumer cannot be null?